relayer/src/supervisor/spawn.rs

```diff
@@ -514,22 +508,48 @@ impl<'a> SpawnContext<'a> {
                 )
                 .then(|| debug!("spawned Client worker: {}", client_object.short_name()));
 
-            // create the Packet object and spawn worker
-            let path_object = Object::Packet(Packet {
-                dst_chain_id: counterparty_chain.id(),
-                src_chain_id: chain.id(),
-                src_channel_id: channel.channel_id,
-                src_port_id: channel.port_id,
-            });
+            let counterparty_channel = IdentifiedChannelEnd::new(
+                channel.channel_end.counterparty().port_id.clone(),
+                channel
+                    .channel_end
+                    .counterparty()
+                    .channel_id
+                    .clone()
+                    .unwrap(),
+                counterparty_channel.unwrap(), // safe because chan_state_dst is Open
+            );
```

Comment:
Wondering if we should change the `channel_on_destination()` to return `Option<IdentifiedChannelEnd>` instead of rebuilding it here.

Reply:
Good idea, I like it! did it here 9a3b4f2c